### ISSUE TITLE
[<Ray component: Core|RLlib|etc...>] 

### ISSUE THREAD

**Issue by gershonavi** [ray-project/ray#32618]
### What happened + What you expected to happen

can not exe bash script using client.submit_job
can not change to bash.

### Versions / Dependencies

aiohttp==3.8.4
aiohttp-cors==0.7.0
aiosignal==1.3.1
anyio==3.6.2
appdirs @ file:///home/conda/feedstock_root/build_artifacts/appdirs_1603108395799/work
argon2-cffi==21.3.0
argon2-cffi-bindings==21.2.0
arrow==1.2.3
asttokens==2.2.1
async-timeout==4.0.2
attrs==22.2.0
backcall==0.2.0
backports.zoneinfo @ file:///home/conda/feedstock_root/build_artifacts/backports.zoneinfo_1667158851884/work
beautifulsoup4==4.11.2
bleach==6.0.0
blessed==1.20.0
bokeh==2.4.3
brotlipy @ file:///home/conda/feedstock_root/build_artifacts/brotlipy_1666764652625/work
cachetools==5.3.0
Cartopy @ file:///home/conda/feedstock_root/build_artifacts/cartopy_1670926143601/work
catboost==1.1.1
certifi==2022.12.7
cffi @ file:///home/conda/feedstock_root/build_artifacts/cffi_1671179356964/work
charset-normalizer==3.0.1
click==8.1.3
cloudpickle==2.2.1
colorful==0.5.5
comm==0.1.2
contourpy @ file:///home/conda/feedstock_root/build_artifacts/contourpy_1673633650729/work
cryptography @ file:///home/conda/feedstock_root/build_artifacts/cryptography-split_1672672382550/work
cycler @ file:///home/conda/feedstock_root/build_artifacts/cycler_1635519461629/work
dask==2023.1.1
debugpy==1.6.6
decorator @ file:///home/conda/feedstock_root/build_artifacts/decorator_1641555617451/work
defusedxml==0.7.1
distlib==0.3.6
distributed==2023.1.1
dogpile.cache==0.9.2
executing==1.2.0
fastjsonschema==2.16.2
filelock==3.9.0
fonttools @ file:///home/conda/feedstock_root/build_artifacts/fonttools_1666827142353/work
fqdn==1.5.1
frozenlist==1.3.3
fsspec==2023.1.0
future==0.18.3
google-api-core==2.11.0
google-auth==2.16.0
googleapis-common-protos==1.58.0
gpustat==1.0.0
graphviz==0.20.1
greenlet @ file:///home/conda/feedstock_root/build_artifacts/greenlet_1674937645728/work
grpcio==1.51.1
HeapDict==1.0.1
idna @ file:///home/conda/feedstock_root/build_artifacts/idna_1663625384323/work
importlib-metadata==6.0.0
importlib-resources==5.10.2
ipykernel==6.21.1
ipython==8.9.0
ipython-genutils==0.2.0
isoduration==20.11.0
jedi==0.18.2
Jinja2==3.1.2
joblib @ file:///home/conda/feedstock_root/build_artifacts/joblib_1663332044897/work
jsonpointer==2.3
jsonschema==4.17.3
jupyter-events==0.6.3
jupyter_client==8.0.2
jupyter_core==5.2.0
jupyter_server==2.2.1
jupyter_server_terminals==0.4.4
jupyterlab-pygments==0.2.2
kiwisolver @ file:///home/conda/feedstock_root/build_artifacts/kiwisolver_1666805766046/work
lightgbm==3.3.5
locket==1.0.0
lxml @ file:///home/conda/feedstock_root/build_artifacts/lxml_1671013487834/work
MarkupSafe==2.1.2
matplotlib @ file:///home/conda/feedstock_root/build_artifacts/matplotlib-suite_1674078821358/work
matplotlib-inline==0.1.6
mistune==2.0.4
msgpack==1.0.4
multidict==6.0.4
munkres==1.1.4
nbclassic==0.5.1
nbclient==0.7.2
nbconvert==7.2.9
nbformat==5.7.3
nest-asyncio==1.5.6
notebook==6.5.2
notebook_shim==0.2.2
numpy @ file:///home/conda/feedstock_root/build_artifacts/numpy_1675642515540/work
nvidia-ml-py==11.495.46
obspy @ file:///home/conda/feedstock_root/build_artifacts/obspy_1674213150972/work
opencensus==0.11.1
opencensus-context==0.1.3
packaging @ file:///home/conda/feedstock_root/build_artifacts/packaging_1673482170163/work
pandas==1.5.3
pandocfilters==1.5.0
parso==0.8.3
partd==1.3.0
pexpect==4.8.0
pickleshare==0.7.5
Pillow @ file:///home/conda/feedstock_root/build_artifacts/pillow_1675487152289/work
Pint==0.20.1
pkgutil_resolve_name==1.3.10
platformdirs==2.6.2
plotly==5.13.0
polars @ file:///home/conda/feedstock_root/build_artifacts/polars_1673481880868/work/target/wheels/polars-0.15.14-cp37-abi3-linux_x86_64.whl
pooch @ file:///home/conda/feedstock_root/build_artifacts/pooch_1643032624649/work
prodict==0.8.18
prometheus-client==0.13.1
prompt-toolkit==3.0.36
protobuf==4.21.12
psutil==5.9.4
psycopg2==2.9.5
ptyprocess==0.7.0
pure-eval==0.2.2
py-spy==0.3.14
pyarrow==11.0.0
pyasn1==0.4.8
pyasn1-modules==0.2.8
pycparser @ file:///home/conda/feedstock_root/build_artifacts/pycparser_1636257122734/work
pydantic==1.10.5
Pygments==2.14.0
pyOpenSSL @ file:///home/conda/feedstock_root/build_artifacts/pyopenssl_1672659226110/work
pyparsing @ file:///home/conda/feedstock_root/build_artifacts/pyparsing_1652235407899/work
pyproj @ file:///home/conda/feedstock_root/build_artifacts/pyproj_1673965689146/work
pyrsistent==0.19.3
pyserial==3.5
pyshp @ file:///home/conda/feedstock_root/build_artifacts/pyshp_1659002966020/work
PySocks @ file:///home/conda/feedstock_root/build_artifacts/pysocks_1661604839144/work
python-dateutil @ file:///home/conda/feedstock_root/build_artifacts/python-dateutil_1626286286081/work
python-json-logger==2.0.4
pytorch-ranger==0.1.1
pytz==2022.7.1
PyVISA==1.13.0
PyYAML==6.0
pyzmq==25.0.0
ray==2.2.0
requests @ file:///home/conda/feedstock_root/build_artifacts/requests_1673863902341/work
rfc3339-validator==0.1.4
rfc3986-validator==0.1.1
rsa==4.9
scikit-learn @ file:///home/conda/feedstock_root/build_artifacts/scikit-learn_1674598999530/work
sciparse==0.1.24
scippy==0.1.31
scipy==1.10.0
seaborn==0.12.2
Send2Trash==1.8.0
shapely @ file:///home/conda/feedstock_root/build_artifacts/shapely_1675079591929/work
six @ file:///home/conda/feedstock_root/build_artifacts/six_1620240208055/work
sklearn==0.0.post1
smart-open==6.3.0
sniffio==1.3.0
sortedcontainers==2.4.0
soupsieve==2.3.2.post1
SQLAlchemy==2.0.2
stack-data==0.6.2
tblib==1.7.0
tenacity==8.1.0
termcolor==2.2.0
terminado==0.17.1
threadpoolctl @ file:///home/conda/feedstock_root/build_artifacts/threadpoolctl_1643647933166/work
timer==0.2.2
tinycss2==1.2.1
toolz==0.12.0
torch==1.9.0
torch-optimizer==0.3.0
tornado==6.2
traitlets==5.9.0
typing_extensions @ file:///home/conda/feedstock_root/build_artifacts/typing_extensions_1665144421445/work
unicodedata2 @ file:///home/conda/feedstock_root/build_artifacts/unicodedata2_1667239955025/work
uri-template==1.2.0
urllib3 @ file:///home/conda/feedstock_root/build_artifacts/urllib3_1673452138552/work
virtualenv==20.17.1
wcwidth==0.2.6
webcolors==1.12
webencodings==0.5.1
WebOb==1.8.7
websocket-client==1.5.1
xgboost==1.7.3
xlrd==1.2.0
XlsxWriter==1.4.4
xmltool==1.0.0
yarl==1.8.2
zict==2.2.0
zipp==3.12.1

### Reproduction script

bash_string = f'''#!/bin/bash     
            echo "Shell being used: $0"
            cd {floder_to_exe}
            pwd
           

            '''

with open(f'{floder_to_exe}/runeng.sh', 'a') as f:
    f.write(bash_string)

job_id = client.submit_job(
    # Entrypoint shell command to execute
    entrypoint=f'. {floder_to_exe}/runeng.sh' # "python -c 'import time; print(\"Sleeping...\"); time.sleep(60)'",
     ,
    runtime_env={
        "working_dir": f"{floder_to_exe}/",
    }

)
print(job_id)

def wait_until_status(job_id, status_to_wait_for, timeout_seconds=5):
    start = time.time()
    while time.time() - start <= timeout_seconds:
        status = client.get_job_status(job_id)
        print(f"status: {status}")
        if status in status_to_wait_for:
            break
        time.sleep(1)

wait_until_status(job_id, {JobStatus.SUCCEEDED, JobStatus.STOPPED, JobStatus.FAILED})
logs = client.get_job_logs(job_id)
print(logs)

### Issue Severity

None

**Comment by jjyao** [comment]
Hi @gershonavi, Could you add more details to the issue you are facing (e.g. what's the error message you are seeing)? Also could you add an issue title?

**Comment by cadedaniel** [comment]
Hi, can you share the failure message you see?
